lib/segment/src/types.rs

```diff
@@ -1196,6 +1196,32 @@ impl GeoRadius {
     }
 }
 
+/// Geo filter request
+///
+/// Matches coordinates inside the polygon, defined by the given coordinates in order.
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub struct GeoPolygon {
+    /// Ordered list of coordinates representing the vertices of a polygon.
+    pub points: Vec<GeoPoint>,
```

Comment:
Something to keep in mind, why I do not want to expose right away this type on the API.

The GeoJSON specs says

```
   To specify a constraint specific to Polygons, it is useful to
   introduce the concept of a linear ring:

   o  A linear ring is a closed LineString with four or more positions.

   o  The first and last positions are equivalent, and they MUST contain
      identical values; their representation SHOULD also be identical.

   o  A linear ring is the boundary of a surface or the boundary of a
      hole in a surface.

   o  A linear ring MUST follow the right-hand rule with respect to the
      area it bounds, i.e., exterior rings are counterclockwise, and
      holes are clockwise.
```

This means we will have a validation layer to ensure those constraints are valid.

It seems the underlying geo crates handle both open & closed polygon transparently by closing the open ones on the fly.